qiskit_nature/drivers/second_quantization/gaussiand/gaussiandriver.py

```diff
@@ -49,69 +49,83 @@ def __init__(
 
         if isinstance(config, list):
             config = "\n".join(config)
 
-        super().__init__(
-            molecule=molecule,
-            basis=basis,
-            method=method.value,
-            supports_molecule=True,
-        )
         self._config = config
 
     @staticmethod
-    def _check_valid():
-        check_valid()
+    def from_molecule(
+        molecule: Molecule,
+        basis: str = "sto-3g",
+        method: MethodType = MethodType.RHF,
+        driver_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> "GaussianDriver":
+        """
+        Args:
+            molecule: molecule
+            basis: basis set
+            method: Hartree-Fock Method type
+            driver_kwargs: kwargs to be passed to driver
+        Returns:
+            driver
+        Raises:
+            QiskitNatureError: Unknown unit
+        """
+        # Ignore kwargs parameter for this driver
+        del driver_kwargs
+        GaussianDriver.check_installed()
+        basis = GaussianDriver.to_driver_basis(basis)
 
-    def _from_molecule_to_str(self) -> str:
-        units = None
-        if self.molecule.units == UnitsType.ANGSTROM:
+        if molecule.units == UnitsType.ANGSTROM:
             units = "Angstrom"
-        elif self.molecule.units == UnitsType.BOHR:
+        elif molecule.units == UnitsType.BOHR:
             units = "Bohr"
         else:
-            raise QiskitNatureError("Unknown unit '{}'".format(self.molecule.units.value))
-        cfg1 = f"# {self.method}/{self.basis} UNITS={units} scf(conventional)\n\n"
-        name = "".join([name for (name, _) in self.molecule.geometry])
+            raise QiskitNatureError("Unknown unit '{}'".format(molecule.units.value))
+        cfg1 = f"# {method.value}/{basis} UNITS={units} scf(conventional)\n\n"
+        name = "".join([name for (name, _) in molecule.geometry])
         geom = "\n".join(
-            [name + " " + " ".join(map(str, coord)) for (name, coord) in self.molecule.geometry]
+            [name + " " + " ".join(map(str, coord)) for (name, coord) in molecule.geometry]
         )
         cfg2 = f"{name} molecule\n\n"
-        cfg3 = f"{self.molecule.charge} {self.molecule.multiplicity}\n{geom}\n\n"
-        return cfg1 + cfg2 + cfg3
+        cfg3 = f"{molecule.charge} {molecule.multiplicity}\n{geom}\n\n"
 
-    def run(self) -> QMolecule:
-        if self.molecule is not None:
-            cfg = self._from_molecule_to_str()
-        else:
-            cfg = self._config
+        return GaussianDriver(cfg1 + cfg2 + cfg3)
+
+    @staticmethod
+    def to_driver_basis(basis: str) -> str:
+        """
+        Converts basis to a driver acceptable basis
+        Args:
+            basis: The basis set to be used
+        Returns:
+            driver acceptable basis
+        """
+        if basis == "sto3g":
+            return "sto-3g"
+        return basis
 
+    @staticmethod
+    def check_installed():
```

Comment:
Since check_valid is `-> None` should the return type here be that too otherwise this almost feels like a boolean. Maybe it should list what it does as a failure with a Raises here? (Same comment I think applies to other places that are similar)

Reply:
I added the typehint return and completed the docstring with the exception